src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -49,22 +49,25 @@ export async function getLambdaHandlerCandidates(uri: vscode.Uri): Promise<Lambd
         (await vscode.commands.executeCommand<vscode.DocumentSymbol[]>('vscode.executeDocumentSymbolProvider', uri)) ??
         []
 
-    return symbols
-        .filter(sym => sym.kind === vscode.SymbolKind.Function)
-        .map<LambdaHandlerCandidate>(symbol => {
-            return {
-                filename,
-                handlerName: `${path.parse(filename).name}.${symbol.name}`,
-                range: symbol.range,
-            }
-        })
+    return symbols.filter(isTopLevelFunction).map<LambdaHandlerCandidate>(symbol => {
+        return {
+            filename,
+            handlerName: `${path.parse(filename).name}.${symbol.name}`,
+            range: symbol.range,
+        }
+    })
+}
+
+function isTopLevelFunction(symbol: vscode.DocumentSymbol) {
+    // if the function is indented at all, it is not classified as being a top-level function in the eyes of Python
```

Comment:
Do we have component level tests around this, that we could add a test for this new "no indent" scenario? 
(I can't remember, not asking you to create them now if they didn't exist)